Automattic/wp-calypso
# Subscription Management: Replace `SiteIcon` component and fix its styling

Fixes https://github.com/Automattic/wp-calypso/issues/78923.

## Proposed Changes

* make all Site icons in Subscription Management squared
* remove the custom Subscription Management's `SiteIcon` component and replace it with the one from `calypso/blocks/site-icon`
* add `alt` attribute to the `SiteIcon` component
* style the default icon using transparent background and `--color-neutral-100` color - to match existing default site icon used in the Reader
* fix styling on the Pending tab

| Before | After |
|--------|--------|
| ![Markup on 2023-07-14 at 10:53:28](https://github.com/Automattic/wp-calypso/assets/25105483/5eff6ec0-83db-47ea-901f-02621e3ce477) | ![Markup on 2023-07-14 at 10:58:59](https://github.com/Automattic/wp-calypso/assets/25105483/9906941c-f333-4f5a-9b62-9eb553dafcb2) |
| ![Markup on 2023-07-14 at 10:54:20](https://github.com/Automattic/wp-calypso/assets/25105483/e80c180c-05e6-45b9-afd3-8cac70a98949) | ![Markup on 2023-07-14 at 10:58:18](https://github.com/Automattic/wp-calypso/assets/25105483/b52308a9-c3c9-4567-8a9d-5b323514075d) |
| ![Markup on 2023-07-14 at 10:56:45](https://github.com/Automattic/wp-calypso/assets/25105483/e05bf5eb-40d7-433b-a88a-a4962ace08fe) | ![Markup on 2023-07-14 at 10:57:38](https://github.com/Automattic/wp-calypso/assets/25105483/dbd5a730-d3f7-45ad-9700-ce439e43ea3b) |
| ![Markup on 2023-07-14 at 11:01:48](https://github.com/Automattic/wp-calypso/assets/25105483/d94c8503-3272-4762-8c51-1f719232997d) | ![Markup on 2023-07-14 at 11:00:26](https://github.com/Automattic/wp-calypso/assets/25105483/904bae34-5e9a-42d6-a009-1e46e34ec642) | 

## Testing Instructions

1. Check out the PR branch and build the app.
2. Make sure your WordPress.com and External users are subscribed to some sites / comments.
3. With the WordPress.com user navigate to http://calypso.localhost:3000/read/subscriptions and confirm all site icons are squared. If the site doesn't have a custom icon, the default dark globe icon should be displayed instead.
4. With the External user navigate to http://calypso.localhost:3000/subscriptions/sites and http://calypso.localhost:3000/subscriptions/comments and confirm all site icons are displayed correctly there as well. You can review the Pending tab (http://calypso.localhost:3000/subscriptions/pending) as well.

ℹ️ To test with the External user on local Calypso, the "subkey cookie dance" is necessary. 🙂 

## Pre-merge Checklist
